webhooks/cluster.go

```diff
@@ -103,3 +143,14 @@ func (c *Cluster) ValidateDelete(ctx context.Context, obj runtime.Object) error
 	}
 	return cluster.ValidateDelete()
 }
+
+// checkClusterClassExists checks for the passed clusterClass in the API server.
+func (c *Cluster) checkClusterClassExists(ctx context.Context, namespace, clusterClass string) *field.Error {
+	if err := c.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterClass}, &clusterv1.ClusterClass{}); err != nil {
```

Comment:
```suggestion
	if err := c.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, &clusterv1.ClusterClass{}); err != nil {
```
nit: maybe we just want to call the parameter `name`.